src/sentry/api/endpoints/integrations/sentry_apps/requests.py

```diff
@@ -68,7 +99,19 @@ def get(self, request: Request, sentry_app) -> Response:
         buffer = SentryAppWebhookRequestsBuffer(sentry_app)
 
         formatted_requests = [
-            self.format_request(req, sentry_app) for req in buffer.get_requests(**kwargs)
+            self.format_request(req, sentry_app, org_slug) for req in buffer.get_requests(**kwargs)
         ]
 
-        return Response(formatted_requests)
+        if start == default_start and end == now:
+            return Response(formatted_requests)
+
+        filtered_requests = []
+        for formatted_request in formatted_requests:
+            if formatted_request.get("date"):
+                date = datetime.strptime(
+                    formatted_request["date"], "%Y-%m-%d %H:%M:%S.%f+00:00"
+                ).replace(microsecond=0)
+                if date >= start and date <= end:
```

Comment:
Also I think it would make sense to filter by date before serialization:
```py
def filter_by_date(request: Mapping[str, Any]) -> bool:
    date_str = request.get("date")
    if not date_str:
        return False
    timestamp = datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S.%f+00:00").replace(microsecond=0)
    return start <= timestamp <= end

class SentryAppRequestsEndpoint(SentryAppBaseEndpoint):
	def get(self, request: Request, sentry_app) -> Response:	
		  ...
	      return Response([
	          RequestSerlializer(request, sentry_app, org_slug).serlialize()
	          for request in SentryAppWebhookRequestsBuffer(sentry_app).get_requests(**kwargs)
	          if filter_by_date(request)
	      ])
```

Reply:
So with the serializer it would look like:
```py
return Response(serialize(filtered_requests, request.user, RequestSerializer(sentry_app)))
```

Reply:
Is `filtered_requests` the result of calling `filter_by_date`? Am I supposed to ignore the previous comment's return?

Reply:
Why do you pass the serializer to the `serialize` function when it takes `obj` (which I'm guessing is the requests that have been filtered), `attrs` (I don't know what that is but it isn't `request.user`), and `user` (which isn't used anywhere in the function)?